sound/soc/sof/ipc4-topology.c

```diff
@@ -1251,7 +1251,7 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 		ipc4_copier = (struct sof_ipc4_copier *)dai->private;
 		copier_data = &ipc4_copier->data;
 		available_fmt = &ipc4_copier->available_fmt;
-		if (dir == SNDRV_PCM_STREAM_CAPTURE) {
+		if (dir == SNDRV_PCM_STREAM_CAPTURE || ipc4_copier->dai_type == SOF_DAI_INTEL_SSP) {
```

Comment:
I am not getting the explanations in the commit message @RanderWang 
Is this really the case that ONLY the SSP gateway assumes 32-bit valid data on playback?

Reply:
And all of @ujfalusi 's comments are valid.

Reply:
@plbossart It is true for closed source FW.  It depends on ssp blob in case of SOF, so I ask @jsarha for help

Reply:
Humm, that doesn't sound good at all. This smells of an undocumented dependency between firmware implementation and NHLT blobs. Meh.

Reply:
yes, different gateway has different requirement and sof should be compatible with closed source fw.